spacelift/resource_run.go

```diff
@@ -80,11 +154,130 @@ func resourceRunCreate(ctx context.Context, d *schema.ResourceData, meta interfa
 		variables["proposed"] = graphql.NewBoolean(graphql.Boolean(proposed.(bool)))
 	}
 
-	if err := meta.(*internal.Client).Mutate(ctx, "ResourceRunCreate", &mutation, variables); err != nil {
+	client := meta.(*internal.Client)
+	if err := client.Mutate(ctx, "ResourceRunCreate", &mutation, variables); err != nil {
 		return diag.Errorf("could not trigger run for stack %s: %v", stackID, internal.FromSpaceliftError(err))
 	}
 
-	d.SetId(mutation.ID)
+	if waitRaw, ok := d.GetOk("wait"); ok {
+		wait := expandWaitConfiguration(waitRaw.([]interface{}))
+
+		if !wait.disabled {
+			stateConf := &retry.StateChangeConf{
+				ContinuousTargetOccurence: 1,
+				Delay:                     10 * time.Second, // TODO: Delay must be a multiple of Timeout
```

Comment:
Did you mean to leave this TODO in?

Reply:
As I see, `Delay` is just a delay before we check the status.
![Screenshot 2024-04-16 at 13 30 26](https://github.com/spacelift-io/terraform-provider-spacelift/assets/114157/e1c51871-c701-4e89-acaa-64cea81903c8)

I've removed that comment.